Feat/add ton jettons

emmanuelm41 opened the pull request:
<!--
Thank you for your contribution! 👍
Please make sure to read CONTRIBUTING.md if you have not already. Pull Requests that do not comply with the rules will be arbitrarily closed.
-->

### ✅ Checklist

<!-- Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready. -->

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** <!-- if not, please explain. (Feature must be tested / Bugfix must bring non-regression) -->
- [ ] **Impact of the changes:** <!-- Please take some time to list the impact & what specific areas Quality Assurance (QA) should focus on -->
  - ...

### 📝 Description

_Replace this text by a clear and concise description of what this pull request is about and why it is needed. Be sure to explain the problem you're addressing and the solution you're proposing._
_For libraries, you can add a code sample of how to use it._
_For bugfixes, you can explain the previous behavior and how it was fixed._
_In case of visual features, please attach screenshots or video recordings to demonstrate the changes._

<!--
| Before        | After         |
| ------------- | ------------- |
|               |               |
-->

### ❓ Context

- **JIRA or GitHub link**: <!-- Attach the relevant ticket number if applicable. (e.g., [JIRA-123] for Jira or #123 for a Github issue) -->


---

### 🧐 Checklist for the PR Reviewers

<!-- Please do not edit this if you are the PR author -->

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

carlosala committed:
feat(ton): add initial definitions

carlosala committed:
feat(ton): add initial definitions

carlosala committed:
feat(ton): add hw-getAddress and hw-signMessage

carlosala committed:
feat(ton): add currencyBridge

carlosala committed:
feat(ton): add accountBridge

carlosala committed:
feat(ton): add LLD integration

carlosala committed:
feat(ton): add LLM integration

carlosala committed:
feat(patches): `buffer` add `Buffer.subarray` polyfill

carlosala committed:
test(ton): add bridge integration test

carlosala committed:
test(ton): add ton specs

carlosala committed:
fix(ton): disable testnet

carlosala committed:
chore: bump pnpm-lock

carlosala committed:
chore: bump pnpm-lock

carlosala committed:
chore: add changeset

ayelenmurano committed:
refac(ton): move ton to coin-framework

ayelenmurano committed:
refac(ton): move ton to coin-framework

ayelenmurano committed:
TO REMOVE enable testnet and currency

ayelenmurano committed:
feat(ton): add initial jetton deffinition

ayelenmurano committed:
TO REMOVE add test jettons

ayelenmurano committed:
--wip-- [skip ci]

ayelenmurano committed:
feat(ton): add sync support jettons

ayelenmurano committed:
fix(ton): display pending operations

ayelenmurano committed:
feat(ton): add jettons data

ayelenmurano committed:
feat(ton): add ability to sign jetton transfers (in process)

ayelenmurano committed:
fix(ton): add a missing import

ayelenmurano committed:
feat(ton): sync sub accounts pending op and adjust token transfers

* The id and hash will be empty, because it will be replaced in broadcast
* Adjust the body of the token transfer
* Replace getSubAccount to findSubAccountById (an existent function)

ayelenmurano committed:
chore(ton): fix a typo

ayelenmurano committed:
feat(lld): Use msg hash to search the tx in the explorer

ayelenmurano committed:
feat(lld): don't allow to send a comment in a token transfer

ayelenmurano committed:
style(ton): lint

ayelenmurano committed:
feat(ton): adjust jettons data

ayelenmurano committed:
chore(ton): Delete unused imports

ayelenmurano committed:
chore(ton): delete comments

emmanuelm41 commented on libs/coin-modules/coin-ton/src/cli-transaction.ts:
```diff
@@ -13,13 +13,18 @@ function inferTransactions(
+  return flatMap(transactions, ({ transaction, account }) => {
     invariant(transaction.family === "ton", "ton family");
 
+    if (account.type === "TokenAccount") {
```
Can we make this a constant?

ayelenmurano replied:
I only need to know if account.type is equal to "TokenAccount" in this moment. Do you think it's necessary?

ayelenmurano replied:
of course, good point!

emmanuelm41 replied:
Great!

emmanuelm41 commented on libs/coin-modules/coin-ton/src/js-estimateMaxSpendable.ts:
```diff
@@ -20,12 +20,32 @@ const estimateMaxSpendable = async ({
   const accountInfo = await fetchAccountInfo(getAddress(a).address);
+
+  if (transaction && !transaction.subAccountId) {
+    transaction.subAccountId = account.type === "Account" ? null : account.id;
```
Same here. Can we make this a constant? or use it from some ledger package if possible,

ayelenmurano replied:
Do you think it's necessary? Because I don't use this value `transaction && !transaction.subAccountId` again

ayelenmurano replied:
I adjusted it to this: `transaction.subAccountId = isTokenAccount(account) ? account.id : null;`. What do you think?

emmanuelm41 replied:
I like it!

emmanuelm41 commented on libs/coin-modules/coin-ton/src/js-prepareTransaction.ts:
```diff
@@ -6,12 +7,22 @@ import { getAddress, getTonEstimatedFees, transactionToHwParams } from "./utils"
+  const tokenTransfer = Boolean(subAccount && subAccount.type === "TokenAccount");
+
+  const simpleTx = transactionToHwParams(t, accountInfo.seqno, a);
+  if (tokenTransfer && simpleTx.payload?.type === "jetton-transfer") {
```
Same here

ayelenmurano replied:
I created a function to know if it's a jetton transfer!

emmanuelm41 replied:
Great!

emmanuelm41 commented on libs/coin-modules/coin-ton/src/js-prepareTransaction.ts:
```diff
@@ -6,12 +7,22 @@ import { getAddress, getTonEstimatedFees, transactionToHwParams } from "./utils"
-  const amount = t.useAllAmount ? a.spendableBalance.minus(t.fees) : t.amount;
+
+  const subAccount = findSubAccountById(a, t.subAccountId || "");
+  const tokenTransfer = Boolean(subAccount && subAccount.type === "TokenAccount");
```
Same here!

ayelenmurano replied:
I replaced `subAccount.type === "TokenAccount"` to `isTokenAccount(subAccount)`

emmanuelm41 replied:
Great!

emmanuelm41 commented on libs/coin-modules/coin-ton/src/js-signOperation.ts:
```diff
@@ -94,4 +75,62 @@ export const buildSignOperation =
+  const { id: accountId } = account;
+
+  const subAccount = findSubAccountById(account, subAccountId || "");
+  const tokenTransfer = subAccount?.type === "TokenAccount";
```
Same here

ayelenmurano replied:
The same here

emmanuelm41 commented on libs/coin-modules/coin-ton/src/js-signOperation.ts:
```diff
@@ -94,4 +75,62 @@ export const buildSignOperation =
+  if (tokenTransfer && subAccount) {
+    op.subOperations = [
+      {
+        id: encodeOperationId(subAccount.id, "", "OUT"),
```
There is an enum for this OUT string. It would be better if we use it.

emmanuelm41 commented on libs/coin-modules/coin-ton/src/js-signOperation.ts:
```diff
@@ -94,4 +75,62 @@ export const buildSignOperation =
+      {
+        id: encodeOperationId(subAccount.id, "", "OUT"),
+        hash: "",
+        type: "OUT",
```
Same here.

emmanuelm41 commented on libs/coin-modules/coin-ton/src/js-signOperation.ts:
```diff
@@ -94,4 +75,62 @@ export const buildSignOperation =
+
+  const value = tokenTransfer ? fees : amount.plus(fees);
+
+  const op: TonOperation = {
+    id: "",
+    hash: "",
+    type: "OUT",
+    senders: [address],
+    recipients: [recipient],
+    accountId,
+    value,
+    fee: fees,
+    blockHash: null,
+    blockHeight: null,
+    date: new Date(),
+    extra: {
+      // we don't know yet, will be patched in final operation
+      lt: "",
+      explorerHash: "",
+      comment: comment,
+    },
+  };
```
Whenever we do a token transfer, this tx does not really exists. I think there is a type "Fee" that you could use instead of OUT, as it is not really an OUT tx, but fees payment

ayelenmurano replied:
mm yes, you are right Emma! I'll add it

ayelenmurano replied:
Hey @emmanuelm41 , I realized I can't change it because the id is created with the type and we receive from Ton "OUT" when it's the fee of the jetton transfer.

emmanuelm41 commented on libs/coin-modules/coin-ton/src/utils.ts:
```diff
@@ -37,19 +38,44 @@ export const transactionToHwParams = (t: Transaction, seqno: number): TonHwParam
+  const subAccount = findSubAccountById(a, t.subAccountId || "");
+
+  const amount = subAccount
+    ? toNano("0.05") // for commission fees, excess will be returned
```
Can we make this a constant somewhere, to make it clear what this value is?

ayelenmurano replied:
I don't find a good name for the constant. What do you think about `maxComissionTokenTransferFee`? @emmanuelm41

emmanuelm41 replied:
could be!

emmanuelm41 commented on libs/coin-modules/coin-ton/src/utils.ts:
```diff
@@ -37,19 +38,44 @@ export const transactionToHwParams = (t: Transaction, seqno: number): TonHwParam
+  const amount = subAccount
+    ? toNano("0.05") // for commission fees, excess will be returned
+    : t.useAllAmount
+    ? BigInt(0)
```
0 as value indicates use all amount on TON?

ayelenmurano replied:
yes, Carlo did in this way, I tested and yes, it worked!

emmanuelm41 replied:
Great!

emmanuelm41 commented on libs/ledgerjs/packages/cryptoassets/src/data/jetton.json:
```diff
@@ -0,0 +1,40 @@
+[
+    ["EQCV4wQkl5dOEhxlicXRY9usvx8c2punEdPROfTpKa45IQnB", "Zondax Test Ton", "ZDXT", 9, false, true],
```
Let's add a TODO here to remove this if it won't be in mainnet.

ayelenmurano replied:
I didn't add it because comments are not allowed in a JSON, but I have the change prepared to delete the testnet changes when the PR is ready. But let me know if I should add the comment in another way

emmanuelm41 replied:
Ok, understood!

emmanuelm41 commented on apps/ledger-live-desktop/cryptoassets.md:
```diff
@@ -14822,6 +14823,8 @@
 | Polygon | WATCH | 0x09211Dc67f9fe98Fb7bBB91Be0ef05f4a12FA2b2 | yieldwatch |  | polygon/erc20/yieldwatch |
 | Stellar | usdc | GA5ZSEJYB37JRC5AVCIA5MOP4RHTM335X2KGX3IHOJAPP5RE34K4KZVN | USDC |  | stellar/asset/USDC:GA5ZSEJYB37JRC5AVCIA5MOP4RHTM335X2KGX3IHOJAPP5RE34K4KZVN |
+| TON | ZDX | EQCstL4MHs8XnWvsiuL4yXJcI_ckHD-4yhUsjEQkTvKqjOU6 | Zondax Test |  | ton/jetton/eqcstl4mhs8xnwvsiul4yxjci_ckhd-4yhusjeqktvkqjou6 |
+| TON | ZDXT | EQCV4wQkl5dOEhxlicXRY9usvx8c2punEdPROfTpKa45IQnB | Zondax Test Ton |  | ton/jetton/eqcv4wqkl5doehxlicxry9usvx8c2punedproftpka45iqnb |
```
Let's add a TODO here to remove this if it won't be in mainnet.

ayelenmurano replied:
It's generated automatically!

emmanuelm41 reviewed (commented):
Pretty solid so far @ayelenmurano!!! It looks great!!

ayelenmurano committed:
refact(ton): use isTokenAccount from coin-framework and clean code

ayelenmurano committed:
fix(ton): Rebase adjustments

ayelenmurano committed:
feat(ton): rebase adjustment

ayelenmurano committed:
feat(ton): add tests and run lint and typecheck commands

ayelenmurano committed:
feat(ton): adjust jetton transfer fees, refactor get transaction status

ayelenmurano committed:
feat(ton): add jetton icons

lawRathod commented on tools/actions/turbo-affected/build/main.js:
```diff
@@ -18595,8 +18595,10 @@ var package_default = {
     "coin:bitcoin": "pnpm --filter coin-bitcoin",
     "coin:evm": "pnpm --filter coin-evm",
     "coin:framework": "pnpm --filter coin-framework",
+    "coin:tester": "pnpm --filter coin-tester",
```
This might be unrelated change...

ayelenmurano replied:
You are right. I'll delete it. Thanks for realizing it!

lawRathod commented on libs/env/src/env.ts:
```diff
@@ -775,8 +775,9 @@ const envDefinitions = {
     desc: "Configure the low battery percentage threshold",
   },
+  // TODO: TON switch to mainnet
   API_TON_ENDPOINT: {
```
We could move this to `config.ts` like in near https://github.com/LedgerHQ/ledger-live/blob/e7ed028716bccb9dc29aa2fc672ecc7a3e78276a/libs/coin-modules/coin-near/src/config.ts

I think they recently added this way of adding env config variables.

ayelenmurano replied:
Really good idea! I hadn't seen it. I'll add the endpoint and key from there.

lawRathod reviewed (approved):
All looks pretty great to me!! 🚀

ayelenmurano committed:
feat(ton): add custom config for ton

emmanuelm41 reviewed (commented):
Looks to me too 🚀

ayelenmurano committed:
add changeset

ayelenmurano committed:
refactor(ton): rebase adjustments

ayelenmurano committed:
feat(ton): Adjust test changes

ayelenmurano committed:
feat(ton): add usdt as jetton

ayelenmurano committed:
refactor(ton): rebase adjustments

ayelenmurano committed:
refactor(ton): solve comments

ayelenmurano committed:
refactor(ton): code style adjustments

ayelenmurano committed:
refactor(ton): solve comments

ayelenmurano committed:
feat(ton): Add NOT as jetton and add MEM icon

ayelenmurano committed:
refactor(ton): add fees payed in the last step, move comment entry to the first one

ayelenmurano committed:
fix(ton): add a missing package and fix an import

ayelenmurano committed:
Add unit tests

ayelenmurano committed:
refactor(ton): solve comments and adjust mobile version

ayelenmurano committed:
refactor(ton): rebase adjustments and adjust jetton decimals

KVNLS committed:
fix: update actions build

ayelenmurano committed:
refactor(ton): Delete a condition

ayelenmurano committed:
refactor(ton): run lint:fix command

ayelenmurano committed:
feat(ton): adjust endpoints to ledger reverse proxy

ayelenmurano committed:
feat(ton): delete jetton integration

rllola committed:
feat(ton): snapshot update

KVNLS committed:
fix: Upgrade pnpm to version 9.4.0

KVNLS committed:
Merge pull request #7248 from LedgerHQ/fix/upgrade-pnpm

Upgrade pnpm to version 9.4.0

abdurrahman-ledger committed:
Revert "ci: change testing firebase env to prod"

This reverts commit 1843b410d735ac848187b004607fb0aa976cc587.

LucasWerey committed:
:boom:feat(lld): nft breadcrumb (#7249)

feat(lld): refactor of breadcrumb for nft gallery

CremaFR committed:
fix: LIVE-13279 (#7256)

* fix: LIVE-13279

* feat: stop caching results from fallback + throws correctly when network error

* chore: changeset

* chore: var renaming

* chore: const in caps

KVNLS committed:
fix: Upgrade Turborepo to 2.0.6

gre committed:
chore(trustchain): add twoAddMembersFollowedByDeviceAdd test (#7229)

chore: add scenario that fails on the trustchain app

hzheng-ledger committed:
Support LIVE-13070 coin tester polkadot runtime upgrade (#7181)

* fix: coin-tester polkadot runtime upgrade

* fix: update changelog

* fix: update polkadot nano app firmware version in coin-tester

* fix: lint error

kallen-ledger committed:
fix: address pr request changes

CremaFR committed:
fix: revert ignoring all swaps error (#7277)

* fix: revert ignoring all swaps error

* chore: changeset

abdurrahman-ledger committed:
Merge pull request #7251 from LedgerHQ/support/playwright_test

[QAA][Playwright] Fix playwright mocked test for features flags prod

abdurrahman-ledger committed:
ci: change back firebase env to testing for speculos

abdurrahman-ledger committed:
Merge pull request #7281 from LedgerHQ/support/qaa-137_speculos_firebase_testing

[QAA-137][Speculos] Change back firebase env to testing

KVNLS committed:
Merge pull request #7276 from LedgerHQ/fix/upgrade-turbo-2

Upgrade Turborepo to 2.0.6

KVNLS committed:
Revert "Merge pull request #7276 from LedgerHQ/fix/upgrade-turbo-2"

This reverts commit ef86807253e481e2d8abad578f870e7423650f2e, reversing
changes made to c34461a51f6f2073bdacafe27f059e79f70d92e9.

KVNLS committed:
Merge pull request #7285 from LedgerHQ/fix/revert-turborepo-2

Revert Turborepo 2

CremaFR committed:
fix: getAvailableProviders returns providers from CAL not from backup (#7278)

* fix: getAvailableProviders returns providers from CAL not from backup

* chore: changeset

kallen-ledger committed:
Merge pull request #7260 from LedgerHQ/bugfix/LIVE-13101

[fix/LIVE-13101]: ensure correct trx fees for trx and trc20 swaps and sends

KVNLS committed:
fix: sign tx for injective

KVNLS committed:
fix: sign tx for injective

CremaFR committed:
feat: checks for tron max values from swap (#7286)

* feat: checks for tron max values from swap

* chore: changeset

* chore: i18n

mcayuelas-ledger committed:
🐛 [BUGFIX] : Reset of market Store when refreshing app + uniformize market store actions (#7259)

live-github-bot[bot] committed:
chore(prerelease) update cryptoassets

live-github-bot[bot] committed:
chore(prerelease): :rocket: entering prerelease mode

lvndry committed:
coin-tester.md

ypolishchuk-ledger committed:
Refactored fixtures file

live-github-bot[bot] committed:
chore(prerelease): :rocket: release prerelease [LLD(2.84.0-next.0), LLM(3.47.0-next.0)]

KVNLS committed:
Merge pull request #7293 from LedgerHQ/fix/zondax-crypto

fix: sign tx for injective

KVNLS committed:
Merge pull request #7295 from LedgerHQ/fix/release-zondax-sign-tx

Release: Fix sign tx for injective

live-github-bot[bot] committed:
chore(prerelease): :rocket: release prerelease [LLD(2.84.0-next.1), LLM(3.47.0-next.1)]

gre committed:
test(trustchain): many getOrCreateTrustchain with different creds fails (#7282)

mcayuelas-ledger committed:
⚡️ [FEAT] : Integrate Trustchain logic in LedgerSync LLD - V1 (#7131)



---------

Co-authored-by: Come Grellard <come.grellard@ledger.fr>
Co-authored-by: Gaëtan Renaudeau <renaudeau.gaetan@gmail.com>

ypolishchuk-ledger committed:
Removing appendFileErrorHandler mathod

mcayuelas-ledger committed:
Analytics WalletSync

sergiubreban committed:
feat: sends devMode to buy-sell app (#7292)

* feat: sends devMode to buy-sell app

* chore: add changeset

abdurrahman-ledger committed:
test: change android firebase environment to testing

mcayuelas-ledger committed:
[FEAT] : Integrate Trustchain logic in LedgerSync LLD - V2 (#7294)

* Rework Add and remove member

* handle error better

* first review

* Rework Deps in Effect

* clean tests

* fix debugger LedgerSYnc for instances

ypolishchuk-ledger committed:
Merge pull request #7289 from LedgerHQ/refactoring/playwright-fixtures

[QAA-133] Refactoring of fixtures file for Playwright

mcayuelas-ledger committed:
Add Close event

KVNLS committed:
fix: upgrade Sentry to V8

mcayuelas-ledger committed:
Add Back event

Commit:
LIVE-13135: fix overflow details on swap device confirmation (#7255)

* fix: overflow details on swap device confirmation

* fix: spacing

* fix: refactor html structure

* fix: add changeset

* fix: drawer scroll

* test(lld): update screenshots (ubuntu-latest)  lld, test, screenshot

* test(lld): update screenshots (ubuntu-latest)  lld, test, screenshot

* test(lld): update screenshots (ubuntu-latest)  lld, test, screenshot

* test(lld): update screenshots (ubuntu-latest)  lld, test, screenshot

* test(lld): update screenshots (ubuntu-latest)  lld, test, screenshot

---------

Co-authored-by: live-github-bot[bot] <105061298+live-github-bot[bot]@users.noreply.github.com>

KVNLS committed:
fix(lld): use correct Sentry on browser side

mcayuelas-ledger committed:
Merge pull request #7299 from LedgerHQ/feat/ledger-sync-analytics-lld

[FEAT]  : Analytics WalletSync

KVNLS committed:
Merge pull request #7302 from LedgerHQ/fix/live-13321

fix(lld): use correct Sentry on renderer side

thesan committed:
🧹 support(lld): Fix generated icons attributes case (#7264)

* Fix generated icons attributes case

* Add changeset

Justkant committed:
feat(web3hub): main and search screen first iteration
prepared a good template for the other screens
fixed a lot of small details with the headers and lists
started i18n
implemented some integration tests for the screens and logic for fetching

Justkant committed:
fix: use auto instead of fit-content that does not exist on RN

Justkant committed:
chore: add wallet-api as codeowners of web3hub

KVNLS committed:
fix(lld): device selector alignment and text overflow

live-github-bot[bot] committed:
test(lld): update screenshots (ubuntu-latest)  lld, test, screenshot

lvndry committed:
add more doc

KVNLS committed:
Merge pull request #7309 from LedgerHQ/fix/live-13332

[LIVE-13332] LLD - fix device selector alignment and text overflow

Justkant committed:
fix: imageProcessor webview should have a source [LIVE-13334]

Justkant committed:
Merge pull request #7311 from LedgerHQ/bugfix/LIVE-13334

fix: imageProcessor webview should have a source [LIVE-13334]

Justkant committed:
Merge pull request #7311 from LedgerHQ/bugfix/LIVE-13334

fix: imageProcessor webview should have a source [LIVE-13334]

live-github-bot[bot] committed:
chore(prerelease): :rocket: release prerelease [LLD(2.84.0-next.1), LLM(3.47.0-next.2)]

thesan committed:
🧪 support(trustchain): Test refusing member removal on device (#7303)

abdurrahman-ledger committed:
test: change iOS firebase testing

gre committed:
chore(trustchain): introduce TrustchainNotAllowed error (#7313)

gre committed:
test(trustchain): getMembers should be Unauthorized if no longer member (#7236)

mcayuelas-ledger committed:
[FEAT] : Handle Unsecured error for Ledger Sync

mcayuelas-ledger committed:
Merge pull request #7314 from LedgerHQ/feat/ws-unsecured-error

[FEAT] : Handle Unsecured error for Ledger Sync

KVNLS committed:
Merge pull request #7306 from LedgerHQ/fix/upgrade-sentry-8

Upgrade @sentry/node to V8 and @sentry/electron to v5

gre committed:
chore(live-wallet): implement Wallet Sync <> Accounts management (#7126)

gre committed:
chore(walletsync): introduce hooks to be able to restoreTrustchain (#7298)

chore: introduce hooks to be able to restoreTrustchain

jiyuzhuang committed:
fix(llm): filter not work bug

Commit:
LIVE-13094: fix long accounts (#7305)

* fix: add text overflow ellipsis for long accounts

* fix: changeset

mcayuelas-ledger committed:
[FIX]: error handling in getMembers (#7316)

* [FIX]: error handling in getMembers

* [REWORK]: Errors

* Fix bug when cancel device action

gre committed:
chore(trustchain): simplify test folders (#7262)

Justkant committed:
fix(LLD): AccountList not showing the correct name and balance for tokens

gre committed:
chore(web-tools): fix AppCloudSync to let unknown fields pass

Justkant committed:
test: add a case for the token name and balance

ayelenmurano committed:
feat(ton): add fees as suboperation on IN txs and update snapshot

Commit:
feat: swap webview content size listener (#7304)

* feat: swap webview content size listener

* fix: add changeset

* test(lld): update screenshots (ubuntu-latest)  lld, test, screenshot

* refactor: expose webviewStyle instead of setWindowContentSize

* fix: missing webviewStyle prop

* fix: lint checks

---------

Co-authored-by: live-github-bot[bot] <105061298+live-github-bot[bot]@users.noreply.github.com>
Co-authored-by: Kant <quentin.jaccarino@ledger.fr>

CremaFR committed:
Feat: network fee drawer for swap live app (#7296)

* wip

* added instant resolve to allow work on swapLiveApp

* evm select works

* mocking evm tx

* feat: using bridge to get correct estimated values

* feat: getSeed address to estimate and convert Amount to Atomic Value

* mutualise code and added openDrawer boolean param

* feat: customFeeConfig

* fix: tokenAccount can edit fees

* chore: renamed estimatedFees

* chore: refactor + convert estimatedFees

* fix: caching gasOption to not default to medium when asking for fast

* feat: adding continue button and cancel changes if exiting

* chore: removed deadcode

* chore: changeset

* chore: remove dead code

* feat: can edit customFeeConfig with previous values

* chore: lint

* chore: lint

Justkant committed:
Merge pull request #7321 from LedgerHQ/bugfix/CN-142

fix(LLD): AccountList not showing the correct name and balance for tokens [CN-142]

Justkant committed:
Merge pull request #7307 from LedgerHQ/feat/LIVE-13176

feat(web3hub): main and search screen first iteration [LIVE-13176]

Justkant committed:
fix: lld bad import for sentry in renderer

KVNLS committed:
Merge pull request #7326 from LedgerHQ/support/fix-lld-bad-import

fix: lld bad import for sentry in renderer

sprohaszka-ledger committed:
[XTZ] Fix prepareTransaction missing data (#7325)

fix(xtz): prepareTransaction missed info

Signed-off-by: Stéphane Prohaszka <stephane.prohaszka@ledger.fr>

sprohaszka-ledger committed:
[XTZ] Fix prepareTransaction missing data (#7325)

fix(xtz): prepareTransaction missed info

Signed-off-by: Stéphane Prohaszka <stephane.prohaszka@ledger.fr>
(cherry picked from commit 1bf61861ceeba147f37429926d19b9a1a244e76c)

live-github-bot[bot] committed:
chore(prerelease): :rocket: release prerelease [LLD(2.84.0-next.2), LLM(3.47.0-next.3)]

jiyuzhuang committed:
fix(lld): onboarding error when device connected

Justkant committed:
feat(web3hub): add search with mocks [LIVE-13187]
Add new test for search screen
Refactor with ViewModel hooks
Update CODEOWNERS folder for Web3Hub

gre committed:
chore: generalize encryption to encrypt trustchain store (#7280)

- app.accounts is the main field that we use to detect if data is encrypted
- but other fields can be introduced as encrypted too and actions done to set the encryption key acts on all
- also rename trustchainStore to trustchain in the app.json

sprohaszka-ledger committed:
[XTZ] Add unit test for prepareTransaction (#7332)

* feat: add some test

Signed-off-by: Stéphane Prohaszka <stephane.prohaszka@ledger.fr>

* feat: add new test with useAllAmount

Signed-off-by: Stéphane Prohaszka <stephane.prohaszka@ledger.fr>

---------

Signed-off-by: Stéphane Prohaszka <stephane.prohaszka@ledger.fr>

jiyuzhuang committed:
Merge pull request #7318 from LedgerHQ/bugfix/live-13346

[LIVE-13346][LLM][BLE] Device search in BLE during onboarding doesn't filter on the selected device

Justkant committed:
refactor(web3hub): navigators

ypolishchuk-ledger committed:
Merge pull request #7315 from LedgerHQ/support/qaa_136_detox_firebase_testing

[QAA-136][Detox] Change firebase environment to testing

Justkant committed:
test: use roles instead of testIDs

Justkant committed:
refactor: export queryKey if needed for invalidation

LucasWerey committed:
chore(release): update LLD release note v2.84.0 and LLM v.3.4.0

gre committed:
chore(wallet): move to pako compression lib (#7331)

gre committed:
chore(trustchain): detect more cases of trustchain refresh needs (#7320)

sprohaszka-ledger committed:
[COIN] Add lastBlock functions to CoinApi (#7265)

* feat: add integ test for CoinApi

Signed-off-by: Stéphane Prohaszka <stephane.prohaszka@ledger.fr>

* feat: add last block function for alpaca

Signed-off-by: Stéphane Prohaszka <stephane.prohaszka@ledger.fr>

* feat: add integ test for coin api

Signed-off-by: Stéphane Prohaszka <stephane.prohaszka@ledger.fr>

* feat: add polkadot lastBlock function

Signed-off-by: Stéphane Prohaszka <stephane.prohaszka@ledger.fr>

* chore: changeset

Signed-off-by: Stéphane Prohaszka <stephane.prohaszka@ledger.fr>

* fix: lint issue

Signed-off-by: Stéphane Prohaszka <stephane.prohaszka@ledger.fr>

* chore: pr feedback

Signed-off-by: Stéphane Prohaszka <stephane.prohaszka@ledger.fr>

---------

Signed-off-by: Stéphane Prohaszka <stephane.prohaszka@ledger.fr>

jiyuzhuang committed:
fix(llm): fix device setup flow error

ayelenmurano committed:
feat(ton): downgrade expo-random and add deleted icons

LucasWerey committed:
Merge pull request #7333 from LedgerHQ/chores/release-note

chore(release): update LLD release note v2.84.0 and LLM v.3.4.0

abdurrahman-ledger committed:
[QAA_116][Detox] Refactor some elements (#7214)

* test: refactor some elements

* test: improve nftGallery

* test: improve liveApps part

gre committed:
chore(wallet): fixes cloudsync api parser on out-of-sync info nullable (#7336)

gre committed:
chore(walletsync): improve walletsync lib (#7337)

Justkant committed:
Merge pull request #7330 from LedgerHQ/feat/LIVE-13177

feat(web3hub): add search with mocks [LIVE-13187]

mcayuelas-ledger committed:
[FEAT] : Activation Flow LedgerSync (#7324)

* Add FollowInstruction Drawer

* ActivationProcess

* Rework Success arch

* Changeset

* Integration test for ActivationProcess

* Improve Activation Test

live-github-bot[bot] committed:
chore(deps): update electron (patch) (#7340)

chore(deps): update electron

Co-authored-by: live-github-bot[bot] <105061298+live-github-bot[bot]@users.noreply.github.com>

gre committed:
chore(live-network): only retry network for status code that makes sense (#7334)

gre committed:
fix(cloudsync): to apply deleted-data on outdated case (#7341)

ypolishchuk-ledger committed:
Activate disabled tests in speculos folder

kallen-ledger committed:
fix: use .eq comparitor for bignumbers

kallen-ledger committed:
fix: update snapshots

ypolishchuk-ledger committed:
Merge pull request #7342 from LedgerHQ/fix/activate-disabled-tests-speculos

Activate disabled tests in speculos folder

kallen-ledger committed:
Merge pull request #7343 from LedgerHQ/bugfix/LIVE-13379

[Bugfix/LIVE-13379] use .eq comparator function in xtz function

live-github-bot[bot] committed:
chore(prerelease): :rocket: release prerelease [LLD(2.84.0-next.3), LLM(3.47.0-next.4)]

mcayuelas-ledger committed:
[CHORE] : Update TRUSTCHAIN_APP_NAME on both platforms (#7338)

* Add FollowInstruction Drawer

* ActivationProcess

* Rework Success arch

* Changeset

* Integration test for ActivationProcess

* Improve Activation Test

* [CHORE] : Update TRUSTCHAIN_APP_NAME on both platforms

Wozacosta committed:
Update libs/coin-tester/coin-tester.md

Wozacosta committed:
Update libs/coin-tester/coin-tester.md

Wozacosta committed:
Update libs/coin-tester/coin-tester.md

Wozacosta committed:
Update libs/coin-tester/coin-tester.md

Wozacosta committed:
Update libs/coin-tester/coin-tester.md

Wozacosta committed:
Update libs/coin-tester/coin-tester.md

add link to bridges doc

VicAlbr committed:
test: adding new json reporter + update e2e workflow + adding aggregate-xray-report script

liviuciulinaru committed:
fix(live-13345): remove quotes filters on demo0 (#7322)

* fix: remove quotes filters on demo0

chore: changeset

fix: remove Filters on legacy too

fix: remove filters test

* test(lld): update screenshots (ubuntu-latest)  lld, test, screenshot

* fix: lint

* fix: lint

* fix: unsued code

---------

Co-authored-by: live-github-bot[bot] <105061298+live-github-bot[bot]@users.noreply.github.com>

liviuciulinaru committed:
feat(live-13282): spacing based on screen resolution (#7329)

* fix: exchange tab named replaced by swap

fix: Tab font size reduced

adjust margins for cards

chore: rm comments

fix: target account

fix: spacing

test(lld): update screenshots (ubuntu-latest)  lld, test, screenshot

* test(lld): update screenshots (ubuntu-latest)  lld, test, screenshot

---------

Co-authored-by: live-github-bot[bot] <105061298+live-github-bot[bot]@users.noreply.github.com>

Justkant committed:
feat(web3hub): add app screen webview [LIVE-13179]
Also add real fetch calls to new manifest-api v2 preview
Update tests for app screen with mocks

valpinkman committed:
chore(common): fix unimportedrc.json

hedi-edelbloute committed:
Feat/integrate icon (#7075)

* feat: support send receive and create account for icon network

---------

Signed-off-by: Stéphane Prohaszka <stephane.prohaszka@ledger.fr>
Co-authored-by: heirland <adam.heirland@gmail.com>
Co-authored-by: heirland <116485607+heirland@users.noreply.github.com>
Co-authored-by: Mang Can Bang <mangcanbang@gmail.com>
Co-authored-by: wilkyway-qx <wilkyway.qx@gmail.com>
Co-authored-by: liemdotv <dovietliemtv@gmail.com>
Co-authored-by: weeddotv <weeddotv@gmail.com>
Co-authored-by: live-github-bot[bot] <105061298+live-github-bot[bot]@users.noreply.github.com>
Co-authored-by: Stéphane Prohaszka <stephane.prohaszka@ledger.fr>

VicAlbr committed:
ci: adding new Job for Xray in test-ui-e2e workflow + renaming customJsonReporter

VicAlbr committed:
ci: removing if condition on upload-to-xray job

valpinkman committed:
Merge pull request #7349 from LedgerHQ/chore/common-unimported-config-file

chore(common): fix unimportedrc.json

jiyuzhuang committed:
Merge pull request #7327 from LedgerHQ/bugfix/live-13353

[LIVE-13353][LLD] Wrong onboarding redirection between Stax and Flex

kallen-ledger committed:
fix: correct error messages in swap when input blance greater than available

kallen-ledger committed:
chore: remove unused import

ldg-smartling-sa committed:
File apps/ledger-live-desktop/static/i18n/en/app.json was translated to ar-AE locale

ldg-smartling-sa committed:
File apps/ledger-live-desktop/static/i18n/en/app.json was translated to zh-CN locale

ldg-smartling-sa committed:
File apps/ledger-live-desktop/static/i18n/en/app.json was translated to fr-FR locale

ldg-smartling-sa committed:
File apps/ledger-live-desktop/static/i18n/en/app.json was translated to de-DE locale

ldg-smartling-sa committed:
File apps/ledger-live-desktop/static/i18n/en/app.json was translated to ja-JP locale

ldg-smartling-sa committed:
File apps/ledger-live-desktop/static/i18n/en/app.json was translated to ko-KR locale

ldg-smartling-sa committed:
File apps/ledger-live-desktop/static/i18n/en/app.json was translated to pt-BR locale

ldg-smartling-sa committed:
File apps/ledger-live-desktop/static/i18n/en/app.json was translated to ru-RU locale

ldg-smartling-sa committed:
File apps/ledger-live-desktop/static/i18n/en/app.json was translated to es-ES locale

ldg-smartling-sa committed:
File apps/ledger-live-desktop/static/i18n/en/app.json was translated to tr-TR locale

ldg-smartling-sa committed:
File apps/ledger-live-mobile/src/locales/en/common.json was translated to ar-AE locale

ldg-smartling-sa committed:
File apps/ledger-live-mobile/src/locales/en/common.json was translated to zh-CN locale

ldg-smartling-sa committed:
File apps/ledger-live-mobile/src/locales/en/common.json was translated to fr-FR locale

ldg-smartling-sa committed:
File apps/ledger-live-mobile/src/locales/en/common.json was translated to de-DE locale

ldg-smartling-sa committed:
File apps/ledger-live-mobile/src/locales/en/common.json was translated to ja-JP locale

ldg-smartling-sa committed:
File apps/ledger-live-mobile/src/locales/en/common.json was translated to ko-KR locale

ldg-smartling-sa committed:
File apps/ledger-live-mobile/src/locales/en/common.json was translated to pt-BR locale

ldg-smartling-sa committed:
File apps/ledger-live-mobile/src/locales/en/common.json was translated to ru-RU locale

ldg-smartling-sa committed:
File apps/ledger-live-mobile/src/locales/en/common.json was translated to es-ES locale

ldg-smartling-sa committed:
File apps/ledger-live-mobile/src/locales/en/common.json was translated to tr-TR locale

ayelenmurano committed:
feat(ton): solve mobile issue on android

liviuciulinaru committed:
fix(live-13358): reset toAmount on target change (#7351)

* fix: reset toAmount on target change

* chore: changeset

* fix: lint

LucasWerey committed:
Merge pull request #7354 from LedgerHQ/smartling-translations-20240717104300275

Smartling on-demand translation delivery from release

VicAlbr committed:
Merge pull request #7018 from LedgerHQ/support/speculos-Xray

Xray link for playwright tests

kallen-ledger committed:
Merge pull request #7353 from LedgerHQ/bugfix/LIVE-13376

[fix/LIVE-13376]: correct error messages in swap when input balance greater than available

live-github-bot[bot] committed:
chore(prerelease): :rocket: release prerelease [LLD(2.84.0-next.4), LLM(3.47.0-next.5)]

live-github-bot[bot] committed:
chore(deps): update dependency @changesets/cli to v2.27.7 (#7348)

Co-authored-by: live-github-bot[bot] <105061298+live-github-bot[bot]@users.noreply.github.com>

Wozacosta committed:
Merge pull request #7323 from LedgerHQ/feat/live-12319-3rd-party-coin-tester-doc

Feat/live 12319 3rd party coin tester doc

gre committed:
test(cloudsync): improve cipher tests (#7344)

gre committed:
chore(trustchain): on removeMember, closeStream with the device for better UX (#7357)

LucasWerey committed:
chore(release): resolve conflicts

mcayuelas-ledger committed:
🏗️ [CHORE] : Rework new Arch folder (#7359)

* [CHORE] : Rework new Arch folder

* Fix unimported

gre committed:
fix(trustchain): invalid jwt token if trustchain/memberCredentials changes (#7358)

Justkant committed:
Merge pull request #7350 from LedgerHQ/feat/LIVE-13179

feat(web3hub): add app screen webview [LIVE-13179]

gre committed:
LLD integration of Wallet Sync (#7300)

chore: kickoff LLD integration of wallet sync

Justkant committed:
feat(web3hub): add tabs button to main, search and app screens [LIVE-13181]
Added a test for tabs button navigation
Remove some testIDs from the tests and use roles (can be a bit slower but also more correct for accessibility)
Mock buttons from RNGH as it was not working in jest tests
Moved the ReactNavigation declaration merging to BaseNavigator next to the declaration of the type
Added MainComposite for tab sub screens used in web3hub tab navigator
Colocate all navigator types
Refactor BackButton
Fixed header styles and animatedStyles for proper placement and sizing

live-github-bot[bot] committed:
chore(release): :rocket: prepare release [skip ci]

LucasWerey committed:
chore(release): resolve conflicts

LucasWerey committed:
Merge pull request #7371 from LedgerHQ/support/release-merge-conflicts

:rotating_light: Release merge conflicts

Justkant committed:
Merge pull request #7365 from LedgerHQ/feat/LIVE-13181

feat(web3hub): add tabs button to main, search and app screens [LIVE-13181]

LucasWerey committed:
Revert "Merge pull request #7371 from LedgerHQ/support/release-merge-conflicts"

This reverts commit 0937c4365b7c04160202d548468d5e9578c65f01, reversing
changes made to d026c1969c8f5802ec44f7a5565e4a5ae2788e53.

LucasWerey committed:
Merge pull request #7372 from LedgerHQ/revert/lockfile

Revert "Merge pull request #7371 from LedgerHQ/support/release-merge-…

mwiercinska committed:
chore: add common setup errors to readme

mwiercinska committed:
Merge pull request #7374 from LedgerHQ/support/add-common-setup-errors-to-readme

chore: add common setup errors to readme

KVNLS committed:
Fix/test rebuild develop (#7383)

* chore(prerelease) update cryptoassets

* chore(prerelease): :rocket: entering prerelease mode

* chore(prerelease): :rocket: release prerelease [LLD(2.84.0-next.0), LLM(3.47.0-next.0)]

* fix: sign tx for injective

* chore(prerelease): :rocket: release prerelease [LLD(2.84.0-next.1), LLM(3.47.0-next.1)]

* chore(prerelease): :rocket: release prerelease [LLD(2.84.0-next.1), LLM(3.47.0-next.2)]

* chore(prerelease): :rocket: release prerelease [LLD(2.84.0-next.2), LLM(3.47.0-next.3)]

* chore(release): update LLD release note v2.84.0 and LLM v.3.4.0

* fix: use .eq comparitor for bignumbers

* fix: update snapshots

* chore(prerelease): :rocket: release prerelease [LLD(2.84.0-next.3), LLM(3.47.0-next.4)]

* File apps/ledger-live-desktop/static/i18n/en/app.json was translated to ar-AE locale

* File apps/ledger-live-desktop/static/i18n/en/app.json was translated to zh-CN locale

* File apps/ledger-live-desktop/static/i18n/en/app.json was translated to fr-FR locale

* File apps/ledger-live-desktop/static/i18n/en/app.json was translated to de-DE locale

* File apps/ledger-live-desktop/static/i18n/en/app.json was translated to ja-JP locale

* File apps/ledger-live-desktop/static/i18n/en/app.json was translated to ko-KR locale

* File apps/ledger-live-desktop/static/i18n/en/app.json was translated to pt-BR locale

* File apps/ledger-live-desktop/static/i18n/en/app.json was translated to ru-RU locale

* File apps/ledger-live-desktop/static/i18n/en/app.json was translated to es-ES locale

* File apps/ledger-live-desktop/static/i18n/en/app.json was translated to tr-TR locale

* File apps/ledger-live-mobile/src/locales/en/common.json was translated to ar-AE locale

* File apps/ledger-live-mobile/src/locales/en/common.json was translated to zh-CN locale

* File apps/ledger-live-mobile/src/locales/en/common.json was translated to fr-FR locale

* File apps/ledger-live-mobile/src/locales/en/common.json was translated to de-DE locale

* File apps/ledger-live-mobile/src/locales/en/common.json was translated to ja-JP locale

* File apps/ledger-live-mobile/src/locales/en/common.json was translated to ko-KR locale

* File apps/ledger-live-mobile/src/locales/en/common.json was translated to pt-BR locale

* File apps/ledger-live-mobile/src/locales/en/common.json was translated to ru-RU locale

* File apps/ledger-live-mobile/src/locales/en/common.json was translated to es-ES locale

* File apps/ledger-live-mobile/src/locales/en/common.json was translated to tr-TR locale

* fix: correct error messages in swap when input blance greater than available

* chore: remove unused import

* chore(prerelease): :rocket: release prerelease [LLD(2.84.0-next.4), LLM(3.47.0-next.5)]

* chore(release): :rocket: prepare release [skip ci]

* feat(web3hub): add tabs button to main, search and app screens [LIVE-13181]
Added a test for tabs button navigation
Remove some testIDs from the tests and use roles (can be a bit slower but also more correct for accessibility)
Mock buttons from RNGH as it was not working in jest tests
Moved the ReactNavigation declaration merging to BaseNavigator next to the declaration of the type
Added MainComposite for tab sub screens used in web3hub tab navigator
Colocate all navigator types
Refactor BackButton
Fixed header styles and animatedStyles for proper placement and sizing

* chore: add common setup errors to readme

* fix: resolve pnpm lock

---------

Co-authored-by: live-github-bot[bot] <105061298+live-github-bot[bot]@users.noreply.github.com>
Co-authored-by: Lucas Werey <lucas.werey@ledger.fr>
Co-authored-by: Kieran Allen <kieran.allen@ledger.fr>
Co-authored-by: ldg-smartling-sa <133769770+ldg-smartling-sa@users.noreply.github.com>
Co-authored-by: Kant <quentin.jaccarino@ledger.fr>
Co-authored-by: Monika Wiercinska <monika.wiercinska@yld.io>

VicAlbr committed:
test: fixing XLM subAccount test (Token Visible in parent account)

VicAlbr committed:
test: adding scroll method for currency selection in addAccount + fixing SubAccount

liviuciulinaru committed:
feat: always show scrollbar in swap live (#7385)

* fix: always show scrollbar

* chore: changeset

mcayuelas-ledger committed:
💄 [FEAT] : Polishing Success Screen + Add new attribute in store (#7360)

[FEAT] : Polishing Success Screen + Add new attribute in store

Commit:
feat: adds swap feedback form (#7379)

VicAlbr committed:
Merge pull request #7386 from LedgerHQ/support/subAccount_fix

🐛 [Speculos] [LLD] Fixing E2E SubAccount tests

LucasWerey committed:
:bug:fix(lld): word break issue in nft drawer (#7389)

mcayuelas-ledger committed:
[FIX]: Allocation display was overlapping content (#7392)

mcayuelas-ledger committed:
[FEAT] : Improve Market for Backend (#7363)

* [FEAT] : Improve Market for Backend

* Sort ids

* Lowercase in hook

mcayuelas-ledger committed:
[FEAT] : Activation Flow with Trustchain (#7352)

* fix(llm): remove global.crypto in LLM because it's broken

* WIP

* WIP 2

* Clean PR + little adjusments

* handle review

* Fix Device not connected

* Remove useless tracer

---------

Co-authored-by: Gaëtan Renaudeau <renaudeau.gaetan@gmail.com>

hzheng-ledger committed:
Fix: LIVE-12323 Rename crypto.org to cronos POS chain (#7232)

fix: rename crypto.org to cronos POS chain

emmanuelm41 committed:
fix: rebase issues

emmanuelm41 committed:
fix: pnpm lock and podfile.lock
